crates/accelerate/src/euler_one_qubit_decomposer.rs

```diff
@@ -649,12 +664,18 @@ fn complex_phase(x: Complex64) -> f64 {
     x.im.atan2(x.re)
 }
 
+/// Wrap angle into interval [-π,π). If within atol of the endpoint, clamp to -π
 #[inline]
-fn mod_2pi(angle: f64) -> f64 {
+fn mod_2pi(angle: f64, atol: f64) -> f64 {
     // f64::rem_euclid() isn't exactly the same as Python's % operator, but because
     // the RHS here is a constant and positive it is effectively equivalent for
     // this case
-    (angle + PI).rem_euclid(2. * PI) - PI
+    let wrapped = (angle + PI).rem_euclid(2. * PI) - PI;
+    if (wrapped - PI).abs() < atol {
+        -PI
+    } else {
+        wrapped
+    }
```

Comment:
I was going to complain that this only clips near the +pi end, not near -pi, but it's pre-existing in the Python implementation, so we can leave it.